### ISSUE TITLE
Directory structure

### ISSUE THREAD

**Issue by osrf-migration** [osrf/gazebo_experimental#7]
**Original report ([archived issue](https://osrf-migration.github.io/osrf-archived-gh-pages/#!/osrf/gazebo_experimental/issues/7)) by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------
I'm opening this issue so we discuss and come up with the desired file structure for this project. Gazebo uses the following structure:

~~~
.
├── cmake
├── deps
├── doc
├── examples
├── gazebo
│      ├── common
│      |       ├── *.cc
│      |       └── *.hh
│      ├── gui
│      |       ├── *.cc
│      |       └── *.hh
│      ├── math
│      |       ├── *.cc
│      |       └── *.hh
│      ├── msgs
│      |       ├── *.cc
│      |       └── *.hh
│      ├── physics
│      |       ├── *.cc
│      |       └── *.hh
│      ├── rendering
│      |       ├── *.cc
│      |       └── *.hh
│      ├── sensors
│      |       ├── *.cc
│      |       └── *.hh
│      ├── test
│      |       ├── *.cc
│      |       └── *.hh
│      ├── transport
│      |       ├── *.cc
│      |       └── *.hh
│      └── util
│              ├── *.cc
│              └── *.hh
├── interfaces
├── media
├── models
├── plugins
├── test
├── tools
└── worlds
~~~

While most recent projects such as SDFormat and Ignition use this:

~~~
.
├── cmake
├── doc
├── examples
├── include
│     └── ignition
│             └── <library_name>
│                       └── *.hh
├── src
│      └── *.cc
├── test
└── tools
~~~

How about something like this here:

~~~
.
├── cmake
├── doc
├── examples
├── include
│     └── gazebo
│             ├── ecs
│             |         ├── Manager.hh
│             |         ├── System.hh
│             |         ├── Entity.hh
│             |         └── *.hh
│             ├── systems
│             |       ├── physics
│             |       |       └── *.hh
│             |       ├── rendering
│             |       |       └── *.hh
│             |       └── <system_name>
│             |               └── *.hh
│             ├── components
│             |         └── *.hh
│             └── <library_name>
│                       └── *.hh
├── src
│      ├── ecs
│      |         ├── Manager.cc
│      |         ├── System.cc
│      |         ├── Entity.cc
│      |        └── *.cc
│      ├── systems
│      |       ├── physics
│      |       |       └── *.cc
│      |       ├── rendering
│      |       |       └── *.cc
│      |       └── <system_name>
│      |               └── *.cc
│      └── <library_name>
│               └── *.cc
├── test
└── tools
~~~

**Comment by osrf-migration** [comment]
**Original comment by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------

* Edited issue description

**Comment by osrf-migration** [comment]
**Original comment by Shane Loretz (Bitbucket: [Shane Loretz](https://bitbucket.org/%7B656e3311-aad9-45a1-aaf7-b0ee0e84b287%7D/), GitHub: [sloretz](https://github.com/sloretz)).**

----------------------------------------
+1

Would there also be a `src/main.cc` for the gazebo executable?

**Comment by osrf-migration** [comment]
**Original comment by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------
Yup, that sounds good to me

**Comment by osrf-migration** [comment]
**Original comment by Shane Loretz (Bitbucket: [Shane Loretz](https://bitbucket.org/%7B656e3311-aad9-45a1-aaf7-b0ee0e84b287%7D/), GitHub: [sloretz](https://github.com/sloretz)).**

----------------------------------------

* set assignee_account_id to "557058:684383ab-ac95-4859-a350-4a6f41a94a22"
* set assignee to "sloretz (Bitbucket: [sloretz](https://bitbucket.org/sloretz), GitHub: [sloretz](https://github.com/sloretz))"

**Comment by osrf-migration** [comment]
**Original comment by Shane Loretz (Bitbucket: [Shane Loretz](https://bitbucket.org/%7B656e3311-aad9-45a1-aaf7-b0ee0e84b287%7D/), GitHub: [sloretz](https://github.com/sloretz)).**

----------------------------------------
I note system header files are in both `/include/gazebo/systems` above and `/src/systems`. Is the latter supposed to be `.cc` instead?

**Comment by osrf-migration** [comment]
**Original comment by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------

* Edited issue description

**Comment by osrf-migration** [comment]
**Original comment by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------
Oh sorry that was a copy/paste error

**Comment by osrf-migration** [comment]
**Original comment by Shane Loretz (Bitbucket: [Shane Loretz](https://bitbucket.org/%7B656e3311-aad9-45a1-aaf7-b0ee0e84b287%7D/), GitHub: [sloretz](https://github.com/sloretz)).**

----------------------------------------
I think the copy/paste error might be a good idea. The ECSystem headers don't need to be included by anyone since they're loaded as plugins. How about putting them next to the source files?

**Comment by osrf-migration** [comment]
**Original comment by Shane Loretz (Bitbucket: [Shane Loretz](https://bitbucket.org/%7B656e3311-aad9-45a1-aaf7-b0ee0e84b287%7D/), GitHub: [sloretz](https://github.com/sloretz)).**

----------------------------------------
I think this is resolved now that [pull request #13](https://osrf-migration.github.io/osrf-archived-gh-pages/#!/osrf/gazebo_experimental/pull-requests/13) has been merged

**Comment by osrf-migration** [comment]
**Original comment by Shane Loretz (Bitbucket: [Shane Loretz](https://bitbucket.org/%7B656e3311-aad9-45a1-aaf7-b0ee0e84b287%7D/), GitHub: [sloretz](https://github.com/sloretz)).**

----------------------------------------

* changed state from "new" to "resolved"

[pull request #13](https://osrf-migration.github.io/osrf-archived-gh-pages/#!/osrf/gazebo_experimental/pull-requests/13)
